src/loaders.py

```diff
@@ -76,7 +80,65 @@ def __getitem__(self, index):
             return Xi
 
     def load_from_memory(self, data_source, index):
-        return data_source[0][0][index]
+        return data_source[0][index]
+
+    def load_from_disk(self, data_source, index):
+        img_filepath = data_source[index]
+        return self.load_image(img_filepath)
+
+    def load_image(self, img_filepath):
+        image = Image.open(img_filepath, 'r')
+        return image.convert('RGB')
+
+
+class ImageSegmentationTTADataset(Dataset):
+    def __init__(self, X, tta_params,
+                 image_transform, image_augment_with_target,
+                 mask_transform, image_augment,
+                 image_source='memory'):
+        super().__init__()
+        self.X = X
+
+        self.image_transform = image_transform
+        self.mask_transform = mask_transform
+        self.image_augment = image_augment if image_augment is not None else ImgAug(iaa.Noop())
+        self.image_augment_with_target = image_augment_with_target if image_augment_with_target is not None else ImgAug(iaa.Noop())
+
+        self.image_source = image_source
+        self.tta_params = tta_params
+
+    def __len__(self):
```

Comment:
@taraspiotr `__len__` is the same as in ImageSegmentationDataset, as well as `load_from_memory`, `load_from_disk` etc methods. Maybe we want to have a base class and we will inherit from it? What do you think @jakubczakon ?

Reply:
@apyskir @taraspiotr I was about to suggest that especially since now we are working on _pre_transform() and _post_transform() methods so yeah I really like the idea.

Reply:
@apyskir @jakubczakon I think `BaseDataset` class can be implemented with these new loaders and just for TTA we can do it like in mapping so that TTA classes inherit from 'normal' ones. What do you think?